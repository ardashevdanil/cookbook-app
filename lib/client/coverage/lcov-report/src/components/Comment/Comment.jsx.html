<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Comment/Comment.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/Comment</a> Comment.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/2</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import PropTypes from 'prop-types';
import React from 'react';
import './Comment.css';
&nbsp;
function Comment({ avatar, date, likes, userName, text }) {
  return (
    &lt;div className="Comment"&gt;
      &lt;div className="Comment__user-info"&gt;
        &lt;img
          alt="User avatar"
          className="Comment__avatar"
          src={avatar}
        /&gt;
        &lt;div className="Comment__info"&gt;
          post by
          &lt;div className="Comment__user-name"&gt;
            {userName}
          &lt;/div&gt;
          &lt;div className="Comment__date"&gt;
            {date}
          &lt;/div&gt;
        &lt;/div&gt;
        &lt;div className="Comment__likes" /&gt;
      &lt;/div&gt;
      &lt;div className="Comment__text"&gt;
        {text}
      &lt;/div&gt;
    &lt;/div&gt;
  );
}
&nbsp;
export default Comment;
&nbsp;
Comment.propTypes = {
  avatar: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  likes: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
  userName: PropTypes.string.isRequired,
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Nov 11 2018 20:41:16 GMT+0300 (MSK)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
