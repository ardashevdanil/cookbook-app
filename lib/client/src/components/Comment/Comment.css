.Comment {
  border-bottom: 2px solid #ddd;
  padding: 20px 10px 0;
}

.Comment__user-info {
  display: flex;
  justify-content: space-between;
}

.Comment__avatar {
  height: 57px;
  width: 57px;
  border-radius: 100%;
  margin-right: 20px;
  background-color: pink;
}

.Comment__info {
  width: calc(100% - 57px - 20px - 50px);
  font-size: 15px;
}

.Comment__user-name {
  margin-top: 2px;
  font-weight: bold;
}

.Comment__date {
  margin-top: 3px;
  font-size: 12px;
  font-weight: bold;
  color: #629494;
}

.Comment__likes {
  height: 0px;
  width: 50px;
  margin-top: 20px;
  font-size: 13px;
  font-weight: bold;
  text-align: end;
}

.Comment__likes>svg {
  font-size: 11px;
}

.Comment__likes:hover {
  color: #629494;
  cursor: pointer;
}
.Comment__text {
  margin: 10px 77px;
  font-size: 14.8px;
}

@media only screen and (max-width: 1120px) {
  .Comment{
    padding: 7vw 3vw;
    border-width: 0.7vw;
  }

  .Comment__avatar {
    height: 18vw;
    width: 18vw;
    margin-right: 5vw;
  }

  .Comment__info {
    width: calc(100% - 18vw - 8vw - 5vw);
    font-size: 4.2vw;
  }

  .Comment__date {
    font-size: 4vw;
  }

  .Comment__likes {
    width: 8vw;
    margin-top: 6vw;
    font-size: 4vw;
  }

  .Comment__likes>svg {
    font-size: 3.4vw;
  }
  
  .Comment__text {
    margin: 3vw 24vw;
    font-size: 4.6vw;
  }
}
