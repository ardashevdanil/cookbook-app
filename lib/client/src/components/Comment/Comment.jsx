import PropTypes from 'prop-types';
import React from 'react';
import './Comment.css';

function Comment({ avatar, date, likes, userName, text }) {
  return (
    <div className="Comment">
      <div className="Comment__user-info">
        <img
          alt="User avatar"
          className="Comment__avatar"
          src={avatar}
        />
        <div className="Comment__info">
          post by
          <div className="Comment__user-name">
            {userName}
          </div>
          <div className="Comment__date">
            {date}
          </div>
        </div>
        <div className="Comment__likes" />
      </div>
      <div className="Comment__text">
        {text}
      </div>
    </div>
  );
}

export default Comment;

Comment.propTypes = {
  avatar: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  likes: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
  userName: PropTypes.string.isRequired,
};
