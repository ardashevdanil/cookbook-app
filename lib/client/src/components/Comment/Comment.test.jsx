import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

import Comment from './Comment';

Enzyme.configure({ adapter: new Adapter() });

describe('test Comment component', () => {
  const component = Enzyme.shallow(
    <Comment
      avatar="test"
      date="test"
      likes={42}
      userName="test"
      text="test"
    />,
  );

  test('renders correctly', () => {
    expect(component.debug()).toMatchSnapshot();
  });
});
