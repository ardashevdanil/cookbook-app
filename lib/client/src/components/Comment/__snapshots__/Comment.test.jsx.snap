// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Comment component renders correctly 1`] = `
"<div className=\\"Comment\\">
  <div className=\\"Comment__user-info\\">
    <div className=\\"Comment__avatar\\" />
    <div className=\\"Comment__info\\">
      post by
      <div className=\\"Comment__user-name\\" />
      <div className=\\"Comment__date\\" />
    </div>
    <div className=\\"Comment__likes\\">
      <FontAwesomeIcon icon={{...}} border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} transform={{...}} />
    </div>
  </div>
  <div className=\\"Comment__text\\" />
</div>"
`;
